### ISSUE TITLE
[BUG][code-analyzer] Scanner ignores <include> <exclude> instrctions in Custom PMD ruleset

### ISSUE THREAD

**Issue by rsoesemann** [forcedotcom/code-analyzer#1928]
### Have you tried to resolve this issue yourself first?

- [x] I confirm I have gone through the above steps and still have an issue to report.

### Bug Description

@johnbelosf asked me to create a bug for that

I am running 

`sf code-analyzer run --rule-selector PMD:OpinionatedSalesforce --output-file code-analyzer-cleancode.csv`

with this PMD ruleset XML

`<?xml version="1.0"?>
<ruleset name="OpinionatedSalesforce" xmlns="http://pmd.sourceforge.net/ruleset/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">

    <description>Aquiva Default PMD Ruleset</description>

    
    <include-pattern>.*/force-app/main/default/.*</include-pattern>

      
    
    <rule ref="category/apex/security.xml">
        <priority>1</priority>
            
        <exclude name="ApexCRUDViolation" />    
    </rule>  `

and it still runs PMD on external libraries in folders outside of `.*/force-app/main/default/.*`

### Output / Logs

```shell

```

### Steps To Reproduce

See above

### Expected Behavior

Not search for issues out of that folder

### Operating System

Mac Seqoia 15.6

### Salesforce CLI Version

2.108.6

### Code Analyzer Plugin (code-analyzer) Version

 5.5.0

### Urgency

Low
